ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -729,11 +731,29 @@ create_attributes_from_template(struct obj_attrs **out, void *template,
 		}
 	}
 
+	/*
+	 * Check and remove duplicates if any and create a new temporary
+	 * template
+	 */
 	rc = sanitize_client_object(&temp, template, template_size, class,
 				    type);
 	if (rc)
 		goto out;
 
+	/*
+	 * For function type modify and copy return the created template
+	 * from here. Rest of the code below is for creating objects
+	 * or generating keys.
+	 */
+	switch (function) {
+	case PKCS11_FUNCTION_MODIFY:
+	case PKCS11_FUNCTION_COPY:
+		*out = temp;
+		return rc;
+	default:
+		break;
+	}
+
```

Comment:
Adding comment to this spot as cannot add it below.

But this is about PKCS11_CKA_LOCAL. If I am not mistaken the current code causes local to be false always?

```
CKA_LOCAL       CK_TRUE only if key was either
...
 created with a C_CopyObject call as a copy of a key which had its CKA_LOCAL attribute set to CK_TRUE
```

Reply:
For function type PKCS11_FUNCTION_MODIFY and PKCS11_FUNCTION_COPY, we have returned back before adding CKA_LOCAL in the template which would be used for modification.

The CKAL_LOCAL value of the parent object (in case of Copy) will be retained during copying.